include/gsl/gsl-lite.hpp

```diff
@@ -1085,13 +1090,12 @@ struct is_compatible_container : std17::bool_constant
 
 template<
     class C, class E
-        , typename = typename std::enable_if<
+        gsl_REQUIRES_T_((
```

Comment:
Note that SFINAE via default template type arguments is fine in *class definitions*. (The two subsequent arguments `class = decltype( ... )` also use default template type arguments for SFINAE.), Our [SFINAE Guidelines](https://github.com/gsl-lite/gsl-lite/blob/master/include/gsl/gsl-lite.hpp#L598..L601) say that the `gsl_REQUIRES_*_()` macros should be avoided "if language support can be assumed in the given context", which is the case here.

Reply:
Commenting at the toplevel.